app/survey_config/census_config.py

```diff
@@ -57,12 +57,17 @@ class CensusSurveyConfig(
         ],
         compare=False,
     )
-    data_layer: Iterable[Mapping] = field(
-        default_factory=lambda: [{"nisra": False}], compare=False
-    )
+
     survey_title: LazyString = lazy_gettext("Census 2021")
     sign_out_button_text: str = lazy_gettext("Save and complete later")
 
+    def get_data_layer(  # pylint: disable=unused-argument, no-self-use
+        self, tx_id=None
+    ) -> list[dict]:
+        if tx_id:
+            return [{"nisra": False}, {"tx_id": tx_id}]
+        return [{"nisra": False}]
```

Comment:
If you assign an attribute call `self._is_nisra` and set the value in each class, then this methods doesn't need to be defined in each class.

Reply:
I've shortened it to list comprehension and only single call, cheers!